tests/calculators/refl1d/test_refl1d_wrapper.py

```diff
@@ -276,3 +282,137 @@ def test_sld_profile(self):
         p.add_item('Item', 'MyModel')
         assert_almost_equal(p.sld_profile('MyModel')[1][0], 0)
         assert_almost_equal(p.sld_profile('MyModel')[1][-1], 4)
+
+
+def test_get_oversampling():
+    # When
+    q = np.linspace(1, 10, 10)
+    dq = np.linspace(0.01, 0.1, 10)
+
+    # Then
+    oversampling = _get_oversampling_q(q_array=q, dq_array=dq, oversampling_factor=5)
+
+    # Expect
+    assert len(oversampling) == 50
+    assert oversampling[0] == 0.965
+    assert oversampling[-1] == 10.35
+
+
+def test_get_probe():
+    # When
+    q = np.linspace(1, 10, 10)
+    dq = np.linspace(0.01, 0.1, 10)
+    model_name = 'model_name'
+
+    storage = {'model': {model_name: {}}}
+    storage['model'][model_name]['scale'] = 10.0
+    storage['model'][model_name]['bkg'] = 20.0
+
+    # Then
+    probe = _get_probe(q_array=q, dq_array=dq, model_name=model_name, storage=storage)
+
+    # Then
+    assert all(probe.Q == q)
+    assert all(probe.calc_Qo == q)
+    assert all(probe.dQ == dq)
+    assert len(probe.calc_Qo) == len(q)
+    assert probe.intensity.value == 10
+    assert probe.background.value == 20
+
+
+def test_get_probe_oversampling():
+    # When
+    q = np.linspace(1, 10, 10)
+    dq = np.linspace(0.01, 0.1, 10)
+    model_name = 'model_name'
+
+    storage = {'model': {model_name: {}}}
+    storage['model'][model_name]['scale'] = 10.0
+    storage['model'][model_name]['bkg'] = 20.0
+
+    # Then
+    probe = _get_probe(q_array=q, dq_array=dq, model_name=model_name, storage=storage, oversampling_factor=2)
+
+    # Then
+    assert len(probe.calc_Qo) == 2 * len(q)
+
+
+def test_get_polarized_probe():
+    # When
+    q = np.linspace(1, 10, 10)
+    dq = np.linspace(0.01, 0.1, 10)
+    model_name = 'model_name'
+
+    storage = {'model': {model_name: {}}}
+    storage['model'][model_name]['scale'] = 10.0
+    storage['model'][model_name]['bkg'] = 20.0
+
+    # Then
+    probe = _get_polarized_probe(q_array=q, dq_array=dq, model_name=model_name, storage=storage)
+
+    # Then
+    assert all(probe.Q == q)
+    assert all(probe.calc_Qo == q)
+    assert all(probe.dQ == dq)
+    assert len(probe.calc_Qo) == len(q)
```

Comment:
I think this is covered by `assert all(probe.calc_Qo == q)`

Reply:
removed:
 assert len(probe.calc_Qo) == len(q)